src/routes/root.rs

```diff
@@ -1,44 +1,27 @@
-use std::default::default;
-
 use afire::prelude::*;
 
-use super::html;
-
-
 pub struct Root;
 
 impl Middleware for Root {
-    fn pre(&self, req: &error::Result<Request>) -> MiddleRequest {
+    fn pre(&self, req: &mut Request) -> MiddleResult {
         // filter out all the requests were not interested in
-        let req = match req {
-            Ok(r) => r,
-            Err(_) => return MiddleRequest::Continue,
-        }
-        .clone();
         if req.method != Method::GET {
-            return MiddleRequest::Continue;
+            return MiddleResult::Continue;
         }
 
         let file_type = req.path.split(".").last().unwrap_or("");
-        // dbg!(file_type);
-        let ret = match file_type {
+        let _ret = match file_type {
             // "md" => md::get_req(req),
-            "html" => html::get_req(req),
-            _ => return MiddleRequest::Continue,
+            // "html" => html::get_req(req),
+            _ => return MiddleResult::Continue,
         };
-
-        if let Ok(req) = ret {
-            return MiddleRequest::Add(req);
-        }
-
-        MiddleRequest::Continue
     }
 }
 
 pub fn attach(server: &mut Server) {
-    server.route(Method::GET, "/", |_req| Response {
-        status: 301,
-        headers: vec![Header::from_string("Location: /home").unwrap()],
-        ..default()
+    server.route(Method::GET, "/", |_req| {
+        Response::new()
+            .status(Status::MovedPermanently)
+            .header(HeaderType::Location, "/home")
```

Comment:
Is that a builder conor

Reply:
its always been like that

(since version [0.2.0](https://github.com/Basicprogrammer10/afire/releases/tag/0.2.0) from late 2021)